nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/DeConv2D.java

```diff
@@ -62,17 +62,30 @@ public DeConv2D(SameDiff sameDiff,
         addArgs();
     }
 
-    public DeConv2D(INDArray[] inputs, INDArray[] outputs, DeConv2DConfig config){
+    public DeConv2D(@NonNull INDArray[] inputs, INDArray[] outputs,@NonNull DeConv2DConfig config){
         super(inputs, outputs);
 
         this.config = config;
         addArgs();
     }
 
-    public DeConv2D(@NonNull INDArray input, @NonNull INDArray weights, INDArray bias, INDArray output, @NonNull DeConv2DConfig config){
+    public DeConv2D(@NonNull INDArray input, @NonNull INDArray weights,@NonNull INDArray bias, INDArray output,@NonNull DeConv2DConfig config){
         this(wrapFilterNull(input, weights, bias), wrapOrNull(output), config);
     }
 
+    public DeConv2D(@NonNull INDArray layerInput,@NonNull INDArray weights, @NonNull DeConv2DConfig deConv2DConfig) {
+        this(wrapFilterNull(layerInput, weights), null, deConv2DConfig);
+    }
+
+    public DeConv2D(@NonNull INDArray layerInput,@NonNull INDArray weights,@NonNull INDArray bias,@NonNull DeConv2DConfig deConv2DConfig) {
+        this(wrapFilterNull(layerInput, weights, bias), null, deConv2DConfig);
+    }
+
+    public DeConv2D(@NonNull INDArray inputs,@NonNull DeConv2DConfig deConv2DConfig) {
+        this(wrapFilterNull(inputs),null, deConv2DConfig);
```

Comment:
I think I've already flagged this previously.
Don't use wrapFilterNull on a single input, just `new INDArray[]{inputs}`
It's like `if(inputs == null){ return null; } else { return new INDArray[]{inputs}; }` which is unnecessary given inputs is never null (by definition given the NonNull annotation)